tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```
@@ -910,6 +1039,45 @@ func (az *AzureProvider) getPrivateIP(ctx context.Context) (string, error) {
 	return *privateIp, nil
 }
 
+// disassociate public IP with the ipConfig.
+// This is needed to delete the public IP address as public IP address cannot be deleted without detaching it from
+// the ipConfig.
+func (az *AzureProvider) disassociateIP(ctx context.Context, nicName string) error {
+	interfaceStruct, err := az.nicClient.Get(ctx, az.resourceGroupName, nicName, "")
+	if err != nil {
+		log.Printf("cannot fetch the network interface data of nic %s: %s", nicName, err)
```

You need to return an error here